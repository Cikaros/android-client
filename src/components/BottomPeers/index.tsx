import React from 'react';
import { Platform, StyleSheet, Text, View } from 'react-native';
import { colors } from '../../theme';

interface Props {
  textCount: number;
  textTotal: number;
}

const styles = StyleSheet.create({
  androidHiddenBorder: {
    borderStyle: 'solid',
    borderLeftWidth: 3,
    borderLeftColor: colors.bottomPeersBgColor,
    borderRightWidth: 3,
    borderRightColor: colors.bottomPeersBgColor,
    position: 'absolute',
    width: '100%',
    left: 0,
    right: 0,
    top: 15 + 38,
    bottom: 0,
    zIndex: 1,
  },
  peersHeader: {
    color: 'rgba(0, 0, 0, 0.45)', 
    fontSize: 16,
    lineHeight: 24,
    fontWeight: "500"
  },
  SeparatorBar: {
    width: 100,
    height: 6,
    borderRadius: 3,
    backgroundColor: 'rgba(217, 217, 217, 1)',
    margin: 16
  }
});

const BottomPeers: React.FC<Props> = ({ textCount, textTotal }) => {
  const backgroundColor =
    Platform.OS == 'ios' ? colors.bottomPeersBgColor : 'transparent';

  return (
    <View
      style={{
        alignItems: 'center',
        width: '100%',
        overflow: 'hidden',
        display: 'flex',
        paddingTop: 15,
      }}>
      <View style={[styles.androidHiddenBorder]}></View>
      <View
        style={{
          width: '100%',
          top: 0,
          paddingTop: 6,
          paddingLeft: 3,
          paddingRight: 3,
          shadowOffset: {
            width: 0,
            height: -3,
          },
          shadowColor: '#171717',
          shadowRadius: 3,
          shadowOpacity: 0.1,
          elevation: 3,

          borderTopStartRadius: 35,
          borderTopEndRadius: 35,
          borderBottomStartRadius: 0,
          borderBottomEndRadius: 0,
          borderLeftColor: colors.bottomPeersBgColor,
          backgroundColor: backgroundColor,
          //overflow: 'hidden',
        }}>
        <View
          style={{
            backgroundColor: colors.bottomPeersBgColor,
            alignItems: 'center',
            borderTopStartRadius: 35,
            borderTopEndRadius: 35,
            borderBottomStartRadius: 0,
            borderBottomEndRadius: 0,
            paddingBottom: 15,
          }}>
          <View style={styles.SeparatorBar}></View>
          <Text testID="peerCountText">
            <Text style={[styles.peersHeader, {fontWeight: '700' }]}>
              {textCount} of {textTotal}{' '}
            </Text>
            <Text style={[styles.peersHeader, {fontWeight: '500' }]}>Peers connected</Text>
          </Text>
        </View>
      </View>
    </View>
  );
};

export default BottomPeers;
