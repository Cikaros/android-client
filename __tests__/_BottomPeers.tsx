import React from 'react';
import { shallow } from 'enzyme';
import BottomPeers from '../src/components/BottomPeers';

describe('BottomPeers', () => {
  const props = {
    textCount: 2,
    textTotal: 5,
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<BottomPeers {...props} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('displays the correct peer count', () => {
    const wrapper = shallow(<BottomPeers {...props} />);
    const peerCountText = wrapper
      .findWhere(node => node.prop('testID') === 'peerCountText')
      .first()
      .prop('children')
      .reduce((acc: string, child: { props: { children: any; }; }) => {
        if (typeof child === 'string') {
          return acc + child;
        }
        if (typeof child === 'object' && typeof child.props.children === 'string') {
          return acc + child.props.children;
        }
        return acc;
      }, '');

    expect(peerCountText).toBe('Peers connected');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<BottomPeers {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
