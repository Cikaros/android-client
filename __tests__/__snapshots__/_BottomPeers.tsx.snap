// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPeers matches snapshot 1`] = `
<Component
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "overflow": "hidden",
      "paddingTop": 15,
      "width": "100%",
    }
  }
>
  <Component
    style={
      [
        {
          "borderLeftColor": "#EDEDED",
          "borderLeftWidth": 3,
          "borderRightColor": "#EDEDED",
          "borderRightWidth": 3,
          "borderStyle": "solid",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 53,
          "width": "100%",
          "zIndex": 1,
        },
      ]
    }
  />
  <Component
    style={
      {
        "backgroundColor": "#EDEDED",
        "borderBottomEndRadius": 0,
        "borderBottomStartRadius": 0,
        "borderLeftColor": "#EDEDED",
        "borderTopEndRadius": 35,
        "borderTopStartRadius": 35,
        "elevation": 3,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 6,
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Component
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EDEDED",
          "borderBottomEndRadius": 0,
          "borderBottomStartRadius": 0,
          "borderTopEndRadius": 35,
          "borderTopStartRadius": 35,
          "paddingBottom": 15,
        }
      }
    >
      <Component
        style={
          {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 3,
            "height": 6,
            "margin": 16,
            "width": 100,
          }
        }
      />
      <Component
        testID="peerCountText"
      >
        <Component
          style={
            [
              {
                "color": "rgba(0, 0, 0, 0.45)",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 24,
              },
              {
                "fontWeight": "700",
              },
            ]
          }
        >
          2
           of 
          5
           
        </Component>
        <Component
          style={
            [
              {
                "color": "rgba(0, 0, 0, 0.45)",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 24,
              },
              {
                "fontWeight": "500",
              },
            ]
          }
        >
          Peers connected
        </Component>
      </Component>
    </Component>
  </Component>
</Component>
`;
